plugins/Wallet/js/transactions.js

```diff
@@ -0,0 +1,275 @@
+'use strict';
+
+// Variable to store transaction history
+var transactions = [];
+
+// Criteria by which to view transactions
+var criteria = {
+	startHeight: 0,
+	endHeight: 1000000,
+	startTime: undefined,
+	endTime: undefined,
+	minValue: undefined,
+	maxValue: undefined,
+	currency: [
+		'siacoin',
+		'siafund',
+		'claim',
+		'miner',
+	],
+	inputs: true,
+	outputs: true,
+	confirmedTransactions: true,
+	uncomfirmedTransactions: true,
+	txnId: undefined,
+	address: undefined,
+};
+
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Updating  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// Append a transaction to Transactions list
+function makeTransaction(txn) {
+	var element = $(`
+		<div class='transaction' id='` + txn.transactionid + `'>
+			<div class='currency'></div>
+			<div class='value'></div>
+			<div class='txnid'></div>
+			<div class='time'></div>
+		</div>
+	`);
+
+	// Insert transaction values in UI
+	element.attr('id', txn.transactionid);
+	element.find('.value').html(txn.value + ' S');
+	element.find('.txnid').html(txn.transactionid);
+	var timestamp = new Date(txn.confirmationtimestamp * 1000);
+	var time = timestamp.toLocaleString();
+	element.find('.time').html(time);
+
+	// Set transaction positive or negative
+	if (txn.value > 0) {
+		element.find('.currency').addClass('positive');
+	} else {
+		element.find('.currency').addClass('negative');
+	}
+
+	// Set transaction currency icon
+	var currencyIcon;
+	if (txn.currency === 'miner') {
+		currencyIcon = '<i class=\'fa fa-heartbeat\'></i>';
+	} else if (txn.currency === 'claim') {
+		currencyIcon = '<i class=\'fa fa-percent\'></i>';
+	} else if (txn.currency === 'siafund') {
+		currencyIcon = '<i class=\'fa fa-diamond\'></i>';
+	} else if (txn.currency === 'siacoin'){
+		currencyIcon = '<i class=\'fa fa-usd\'></i>';
+	}
+	element.find('.currency').append(currencyIcon);
+
+	// Add and display transaction
+	$('#transaction-list').append(element);
+}
+
+// Fill address page with search results or addresses
+function updateTransactionPage() {
+	$('#transaction-list').empty();
+
+	// Set page limits
+	$('#transaction-page').attr({
+		min: 1,
+		max: transactions.length === 0 ? 1 : Math.ceil(transactions.length / 25),
+	});
+
+	// Make elements for this page
+	var n = (($('#transaction-page').val() - 1) * 25);
+	transactions.slice(n, n + 25).forEach(function(processedTransaction) {
```

Comment:
is 25 the number of items per page? If so, it should be explicitly defined somewhere, even if it's just at the top of this function. `const pageSize = 25`